src/Symfony/Component/DependencyInjection/Compiler/PassConfig.php

```diff
@@ -42,13 +42,13 @@ public function __construct()
         $this->beforeOptimizationPasses = array(
             100 => array(
                 $resolveClassPass = new ResolveClassPass(),
-                new ResolveDefinitionInheritancePass(),
```

Comment:
I moved this down to be *after* `ResolveDefinitionTemplatesPass` on purpose: this allows the parent-child definitions to be resolved first. Then, the `ResolveDefinitionInheritancePass` doesn't need to worry about this.